### PR TITLE
Add findTestHelpers() to find helpers and fixtures

### PR DESCRIPTION
**Pull request by vadimdemedes**
Used by https://github.com/avajs/ava/pull/1078, so that we can precompile helpers and fixtures as well.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -30,6 +30,10 @@ avaFiles.isSource(filePath);
 avaFiles.findTestFiles().then(files => {
 	// files is an array of found test files
 });
+
+avaFiles.findTestHelpers().then(files => {
+	// files is an array of found test helpers and fixtures
```
Comment (novemberborn) [inline readme.md]: I'd argue that fixtures are different from helpers.

File: index.js [hunk 2]
```diff
@@ -58,6 +58,23 @@ AvaFiles.prototype.findTestFiles = function () {
...
+		cwd: this.cwd,
+		cache: Object.create(null),
+		statCache: Object.create(null),
+		realpathCache: Object.create(null),
+		symlinks: Object.create(null)
+	}).then(function (files) {
+		return files.filter(function (file) {
+			return multimatch([file], [
+				'**/fixtures/**',
+				'**/helpers/**'
```
Comment (novemberborn) [inline index.js]: Helpers may also start with an underscore.

File: readme.md (hunk 1 above)
Comment (vdemedes) [inline readme.md]: From the point of this method's functionality of precompiling code, they're the same. Are you referring to the function name or comment?

File: index.js [hunk 3]
```diff
@@ -58,6 +58,23 @@ AvaFiles.prototype.findTestFiles = function () {
...
+AvaFiles.prototype.findTestHelpers = function () {
+	return handlePaths(this.files, ['!**/node_modules/**'], {
+		cwd: this.cwd,
+		cache: Object.create(null),
+		statCache: Object.create(null),
+		realpathCache: Object.create(null),
+		symlinks: Object.create(null)
+	}).then(function (files) {
+		return files.filter(function (file) {
+			return multimatch([file], [
```
Comment (sindresorhus) [inline index.js]: Why not just pass `files` directly to `multimatch`?

File: test.js [hunk 4]
```diff
@@ -139,3 +139,35 @@ test('findFiles - finds the correct files by default', async t => {
...
+	const fixtureDir = fixture('default-patterns');
+	process.chdir(fixtureDir);
+
+	const expected = [
+		'sub/directory/__tests__/fixtures/foo.js',
+		'sub/directory/__tests__/helpers/foo.js',
+		'test/fixtures/foo-fixt.js',
+		'test/helpers/test.js'
+	].map(function (file) {
+		return path.join(fixtureDir, file);
```
Comment (sindresorhus) [inline test.js]: Use inline arrow function.

Comment (vdemedes) [inline test.js]: Shouldn't it be compatible with older Node.js versions?

Comment (sindresorhus) [inline test.js]: @vdemedes This is the test though, run through Babel with AVA.

Comment (vdemedes) [inline test.js]: Oh, I confused this with the other thing, nvm.

**Comment by vdemedes** [comment]
Updated PR according to all the feedback, thanks guys!

**Comment by sindresorhus** [comment]
LGTM. @novemberborn ?

**Pull request merged**